app/src/main/java/org/wikipedia/diff/ArticleEditDetailsFragment.kt

```diff
@@ -0,0 +1,401 @@
+                        HistoryEntry(articlePageTitle, HistoryEntry.SOURCE_ON_THIS_DAY_ACTIVITY), articlePageTitle))
+            }
+        }
+        newerIdButton.setOnClickListener {
+            revisionId = newerRevisionId
+            fetchEditDetails()
+        }
+        olderIdButton.setOnClickListener {
+            revisionId = olderRevisionId
+            fetchEditDetails()
+        }
+        watchButton.setOnClickListener {
+            watchButton.isCheckable = false
+            watchOrUnwatchTitle(watchlistExpirySession, isWatched)
+        }
+        usernameButton.setOnClickListener {
+            if (AccountUtil.isLoggedIn() && username != null) {
+                startActivity(newIntent(requireActivity(),
+                        PageTitle(UserTalkAliasData.valueFor(languageCode),
+                                username!!, WikiSite.forLanguageCode(languageCode))))
+            }
+        }
+        thankButton.setOnClickListener { showThankDialog() }
+    }
+
+    private fun setUpInitialUI() {
+        diffSize = arguments?.getInt(INTENT_EXTRA_EDIT_SIZE, 0)!!
+        diffText.movementMethod = ScrollingMovementMethod()
+        articleTitleView.text = articlePageTitle.displayText
+        if (diffSize >= 0) {
+            diffCharacterCountView.setTextColor(if (diffSize > 0) ContextCompat.getColor(requireContext(),
+                    R.color.green50) else ResourceUtil.getThemedColor(requireContext(), R.attr.material_theme_secondary_color))
+            diffCharacterCountView.text = String.format("%+d", diffSize)
+        } else {
+            diffCharacterCountView.setTextColor(ContextCompat.getColor(requireContext(), R.color.red50))
+            diffCharacterCountView.text = String.format("%+d", diffSize)
+        }
+    }
+
+    private fun fetchEditDetails() {
+        disposables.add(Observable.zip(ServiceFactory.get(WikiSite.forLanguageCode(languageCode)).getRevisionDetails(articlePageTitle.prefixedText, revisionId),
+                ServiceFactory.get(WikiSite.forLanguageCode(languageCode)).getWatchedInfo(articlePageTitle.prefixedText), { r, w ->
+            isWatched = w.query()!!.firstPage()!!.isWatched
+            if (r.query() == null || r.query()!!.firstPage() == null) {
+                throw RuntimeException("Received empty response page: " + GsonUtil.getDefaultGson().toJson(r))
+            }
+            val firstPage = r.query()!!.firstPage()!!
+            currentRevision = firstPage.revisions()[0]
+            username = currentRevision!!.user
+            newerRevisionId = if (firstPage.revisions().size < 2) {
+                -1
+            } else {
+                firstPage.revisions()[1].revId
+            }
+            olderRevisionId = currentRevision!!.parentRevId
+        })
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    updateUI()
+                }) { t: Throwable? -> L.e(t) })
```

Comment:
Would it be possible to show the `WikiErrorView` in the layout in case something goes wrong?